File: kaizen-openapi-parser/src/main/java/com/reprezen/kaizen/oasparser/model3/ServerVariable.java
package com.reprezen.kaizen.oasparser.model3;

import java.util.List;
import java.util.Map;

import javax.annotation.Generated;

import com.reprezen.jsonoverlay.IJsonOverlay;
import com.reprezen.jsonoverlay.IModelPart;

public interface ServerVariable extends IJsonOverlay<ServerVariable>, IModelPart<OpenApi3, ServerVariable> {

	// EnumValue
	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	List<Object> getEnumValues();

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	List<Object> getEnumValues(boolean elaborate);

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	boolean hasEnumValues();

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	Object getEnumValue(int index);

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	void setEnumValues(List<Object> enumValues);

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	void setEnumValue(int index, Object enumValue);

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	void addEnumValue(Object enumValue);

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	void insertEnumValue(int index, Object enumValue);

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	void removeEnumValue(int index);

	// Default
	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	Object getDefault();

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	void setDefault(Object defaultValue);

	// Description
	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	String getDescription();

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	void setDescription(String description);

	// Extension
	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	Map<String, Object> getExtensions();

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	Map<String, Object> getExtensions(boolean elaborate);

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	boolean hasExtensions();

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	boolean hasExtension(String name);

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	Object getExtension(String name);

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	void setExtensions(Map<String, Object> extensions);

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	void setExtension(String name, Object extension);

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	void removeExtension(String name);
}

File: src/main/java/com/reprezen/kaizen/oasparser/model3/ServerVariable.kt
package com.reprezen.kaizen.oasparser.model3

import com.reprezen.jsonoverlay.parser.Generated;
import com.reprezen.jsonoverlay.IJsonOverlay;
import com.reprezen.jsonoverlay.IModelPart;
import kotlin.collections.List;
import kotlin.collections.Map;

interface ServerVariable : IJsonOverlay<ServerVariable>, IModelPart<OpenApi3, ServerVariable> {

	// EnumValue
	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun getEnumValues() : List<String>

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun getEnumValues(elaborate : Boolean) : List<String>

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun hasEnumValues() : Boolean

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun getEnumValue(index : Int) : String

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun setEnumValues(enumValues : MutableList<String>)

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun setEnumValue(index : Int, enumValue : String)

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun addEnumValue(enumValue : String)

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun insertEnumValue(index : Int,enumValue : String)

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun removeEnumValue(index : Int)

	// Default
	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun getDefault() : String?

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun setDefault(defaultValue : String)

	// Description
	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun getDescription() : String?

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun setDescription(description : String)

	// Extension
	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun getExtensions() : MutableMap<String, Any>

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun getExtensions(elaborate : Boolean) : MutableMap<String, Any>

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun hasExtensions() : Boolean

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun hasExtension(name : String) : Boolean

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun getExtension(name : String) : Any?

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun setExtensions(extensions : MutableMap<String, Any>)

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun setExtension(name : String,extension : Any)

	@Generated("com.reprezen.jsonoverlay.gen.CodeGenerator")
	fun removeExtension(name : String)

}
